src/prebid.js

```diff
@@ -100,20 +102,41 @@ function validateSizes(sizes, targLength) {
   return cleanSizes;
 }
 
-export function setBattrForAdUnit(adUnit, mediaType) {
-  const ortb2Imp = adUnit.ortb2Imp || {};
-  const mediaTypes = adUnit.mediaTypes || {};
+// synchronize fields between mediaTypes[mediaType] and ortb2Imp[mediaType]
+export function syncOrtb2(adUnit, mediaType) {
+  const ortb2Imp = deepAccess(adUnit, `ortb2Imp.${mediaType}`);
+  const mediaTypes = deepAccess(adUnit, `mediaTypes.${mediaType}`);
 
-  if (ortb2Imp[mediaType]?.battr && mediaTypes[mediaType]?.battr && (ortb2Imp[mediaType]?.battr !== mediaTypes[mediaType]?.battr)) {
-    logWarn(`Ad unit ${adUnit.code} specifies conflicting ortb2Imp.${mediaType}.battr and mediaTypes.${mediaType}.battr, the latter will be ignored`, adUnit);
+  if (!ortb2Imp && !mediaTypes) {
+    // omitting sync due to not present mediaType
+    return;
   }
 
-  const battr = ortb2Imp[mediaType]?.battr || mediaTypes[mediaType]?.battr;
+  const fields = {
+    'video': ORTB_VIDEO_PARAMS,
+    'native': ORTB_NATIVE_PARAMS,
```

Comment:
I would not deal with native here, the mediaType and imp fields have no overlap, and it gets complicated with `mediaType.native.ortb`.